indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -185,3 +191,61 @@ class RHRegistrationEdit(RHManageRegistrationBase):
     def _process(self):
         return WPManageRegistration.render_template('management/registration_modify.html', self.event,
                                                     registration=self.registration)
+
+
+class RHRegistrationEmailRegistrants(RHManageRegFormBase):
+
+    def _checkParams(self, params):
+        self._doNotSanitizeFields.append('from_address')
+        RHManageRegFormBase._checkParams(self, params)
+
+    def _get_people(self):
+        ids = set(request.form.getlist('registration_ids'))
+        return Registration.find(Registration.id.in_(ids), ~Registration.is_deleted).with_parent(self.regform).all()
+
+    def _send_emails(self, form):
+        people = self._get_people()
+        for person in people:
+            template = get_template_module('events/registration/emails/custom_email.html',
+                                           email_subject=form.subject.data, email_body=form.body.data)
+            email = make_email(to_list=person.email, cc_list=form.cc_addresses.data,
+                               from_address=form.from_address.data, template=template, html=True)
+            send_email(email, self.event)
+
+    def _process(self):
+        form = EmailRegistrantsForm()
+        if form.validate_on_submit():
+            self._send_emails(form)
+            return jsonify_data(flashed_messages=flash(_("Emails sent"), 'success'))
+        return WPManageRegistration.render_template('management/email.html', form=form)
+
+
+class RHRegistrationDelete(RHManageRegFormBase):
+
+    def _process(self):
+        ids = set(request.form.getlist('registration_ids'))
+        registrations = (Registration.find(Registration.id.in_(ids), ~Registration.is_deleted).with_parent(self.regform)
```

Comment:
i'd wrap it more nicely when you need to wrap anyway. i.e. insert linebreaks before `.find` and `.with_parent`